### ISSUE TITLE
Allow setting the default responseHandler at a higher level

### ISSUE THREAD

**Issue by taylorkline** [reduxjs/redux-toolkit#2385]
Because of #2354 (which is addressed in #2363, pending feedback), we have resorted to adding a custom `responseHandler` to each and every endpoint definition.

This gets tedious, and it's easy for one of the several developers on our team to miss a case until we run into a run-time `PARSING_ERROR` when we get a `"text/plain"` response.

It would be very convenient to be able to set the default `responseHandler` across all endpoints in the options to `createApi`.

**Comment by phryneas** [comment]
A PR adding `responseHandler` to the `FetchBaseQueryArgs` argument to `fetchBaseQuery` would be welcome.

If you are not using TypeScript, it is probably even already possible to pass it into `fetchBaseQuery` directly at creation.

**Comment by markerikson** [comment]
Slightly confused re-reading this, but I _think_ this is fixed by #2363 ?

**Comment by markerikson** [comment]
I take it back. Re-re-reading our own types: I think you can pass `responseHandler` as part of an individual `query` object, but not to `fetchBaseQuery`?

**Comment by taylorkline** [comment]
Short version: it would be nice to be able to specify `responseHandler` in `createApi`:

```ts
export const pokemonApi = createApi({
  reducerPath: 'pokemonApi',
  baseQuery: fetchBaseQuery({ baseUrl: 'https://pokeapi.co/api/v2/' }),
  responseHandler: 'content-type',
  endpoints: (builder) => ({
    getPokemonByName: builder.query<Pokemon, string>({
      query: (name) => `pokemon/${name}`,
    }),
  }),
})
```

Or similar.

**Comment by phryneas** [comment]
I added a PR for that today - #2823

**Comment by markerikson** [comment]
Merged into the 1.9 branch
